### PR TITLE
Update: Specify search field for table selector

### PR DESCRIPTION
**Pull request by ldkge**
Resolves #

## Description
Add the searchField property to the navi-table-select so that you can specify which field you want to use for search when search is enabled.

## Proposed Changes

- Add searchField property.

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/reports/addon/components/navi-table-select.js [hunk 1]
```diff
@@ -7,6 +7,7 @@
  *          selected=selected
  *          onChange=onChange
  *          searchEnabled=searchEnabled
+ *          searchField=searchField
```
Comment (kevinhinterlong) [inline packages/reports/addon/components/navi-table-select.js]: Can we make this a native class, update docs/copyright

File: packages/reports/addon/templates/components/navi-table-select.hbs [hunk 2]
```diff
@@ -5,6 +5,7 @@
   @options={{@options}}
   @selected={{@selected}}
   @searchEnabled={{searchEnabled}}
+  @searchField={{searchField}}
```
Comment (kevinhinterlong) [inline packages/reports/addon/templates/components/navi-table-select.hbs]:
```suggestion
  @searchEnabled={{this.searchEnabled}}
  @searchField={{this.searchField}}
```

**Comment by kevinhinterlong** [comment]
Maybe add a test for searching by `longName`?

File: packages/reports/addon/components/navi-table-select.js [hunk 3]
```diff
@@ -22,5 +23,10 @@ export default Component.extend({
   /**
    * @property {Boolean} searchEnabled
    */
-  searchEnabled: false
+  searchEnabled: false,
```
Comment (kevinhinterlong) [inline packages/reports/addon/components/navi-table-select.js]: Can we change this component to be a native class?

File: packages/reports/tests/integration/components/navi-table-select-test.js [hunk 4]
```diff
@@ -58,15 +58,45 @@ module('Integration | Component | navi table select', function(hooks) {
...
+      'All options are shown'
+    );
+
+    await fillIn('.ember-power-select-search-input', '2');
+    assert.deepEqual(
+      findAll('.ember-power-select-option').map(el => el.textContent.trim()),
+      ['network2'],
+      'Filtered options are shown'
+    );
+  });
```
Comment (colemanProjects) [inline packages/reports/tests/integration/components/navi-table-select-test.js]: can you add a test to clear, or fillIn empty. and all options should be shown again. 

and another test, to make sure if the input doesn't match, the appropriate thing is shown

File: packages/reports/addon/components/navi-table-select.js (hunk 3 above)
Comment (ldkge) [reply packages/reports/addon/components/navi-table-select.js]: what do we mean by native class?

Comment (mattgraham1995) [reply packages/reports/addon/components/navi-table-select.js]: Line 18. `export default Class NaviTableSelect extends Component {`

Comment (kevinhinterlong) [reply packages/reports/addon/components/navi-table-select.js]:
```js
import { layout as templateLayout, tagName } from '@ember-decorators/component';

@templateLayout(layout)
@tagName('')
class NativeTableSelectComponent extends Component {
  /**
   * @property {Boolean} searchEnabled
   */
  searchEnabled = false;

  /**
   * @property {String} searchField
   */
  searchField = 'longName';
}

export default NativeTableSelectComponent;
```

Then in the template
```hbs
<div class="navi-table-select" ...attributes>
  {{!-- rest of template here --}}
</div>
```

Comment (ldkge) [reply packages/reports/addon/components/navi-table-select.js]: ohhh I got what you mean now

File: packages/reports/addon/components/navi-table-select.js [hunk 5]
```diff
@@ -1,26 +1,33 @@
...
- *          searchEnabled=searchEnabled
- *          tagName=tagName
- *        }}
+ * Usage: <NaviTableSelect
+ *          @options={{options}}
+ *          @selected={{selected}}
+ *          @onChange={{onChange}}
+ *          @searchEnabled={{searchEnabled}}
+ *          @searchField={{searchField}}
+ *          @tagName={{tagName}}
```
Comment (jkusa) [inline packages/reports/addon/components/navi-table-select.js]:
```suggestion
```
